models/api-key.js

```diff
@@ -114,6 +161,29 @@ const isAlreadyActivated = (apiKeyRecord) => {
 };
 module.exports.isAlreadyActivated = isAlreadyActivated;
 
+const isValidApiSecret = (apiKeyRecord, apiSecret = '') => {
+  if (!apiKeyRecord) {
```

Comment:
might consider moving these timing-protection lines into compare for cohesion.

Reply:
Sounds good. I may do that as a separate PR just to avoid changing this PR too much after approvals.